### PR TITLE
Fixed ReduceMean layer behaviour

### PR DESCRIPTION
**Pull request by LaurentBerger**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

https://github.com/microsoft/onnxruntime/blob/a93c31e3c9971063d8dfe45a627a80cbdcf99ed9/onnxruntime/core/providers/cpu/reduction/reduction_ops.cc#L433-L443

### CODE REVIEW

**Comment by asmorkalov** [comment]
@LaurentBerger Thanks a lot for the patch! Could you add a test with the case in the ticket. You can use this test as an example: https://github.com/opencv/opencv/blob/5e5a035c5bddbd0190f6cec8e9db90363f402799/modules/dnn/test/test_layers.cpp#L1799

**Review by Abdurrahheem — APPROVED** [review]

File: modules/dnn/test/test_layers.cpp [hunk 1]
```diff
@@ -1795,6 +1795,52 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Test_ShuffleChannel, Combine(
...
+        lp.set("axes", i);
+        lp.set("keepdims", 0);
+        lp.type = "Reduce";
+        lp.set("reduce", "MEAN");
+        lp.name = "testReduceMean";
+        lp.blobs.push_back(inpInitA);
+        net.addLayerToPrev(lp.name, lp.type, lp);
+        net.setInput(inpInitA);
+        net.setPreferableBackend(DNN_BACKEND_OPENCV);
+        Mat output = net.forward();
```
Comment (asmorkalov) [inline modules/dnn/test/test_layers.cpp:1823]: I cannot understand what the test actually does. Why `forward()` is called in a loop, if the network parameters and input is not changed?

Comment (LaurentBerger) [reply modules/dnn/test/test_layers.cpp:1823]: I changed axes parameter

https://github.com/opencv/opencv/blob/c96dd8be93e5627ad30a96b27a85f514bdda85b1/modules/dnn/test/test_layers.cpp#L1814

**Review by asmorkalov — APPROVED** [review]
:+1: Thanks a lot for the contribution.

**Review by fengyuentau — APPROVED** [review]

**Pull request merged**